core/src/main/java/com/digitalpebble/stormcrawler/protocol/selenium/SeleniumGridImpl.java

```diff
@@ -0,0 +1,118 @@
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.digitalpebble.stormcrawler.protocol.selenium;
+
+import com.digitalpebble.stormcrawler.util.ConfUtils;
+import java.net.URL;
+import java.time.Duration;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import org.apache.storm.Config;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+public class SeleniumGridImpl extends SeleniumGridProtocol {
+
+    private TimerTask timerTask;
+    private Timer updateQueue;
+    private int noOfWorkers = 0;
+    private String gridAddress;
+    private final DesiredCapabilities capabilities = new DesiredCapabilities();
+
+    @Override
+    public void configure(Config conf) {
+        super.configure(conf);
+        noOfWorkers = ConfUtils.getInt(conf, "topology.workers", 2);
+        gridAddress = super.gridAddress;
+        capabilities.setBrowserName(
+                ConfUtils.getString(conf, "selenium.capabilities.browserName", "chrome"));
+        capabilities.setCapability("newSessionWaitTimeout", 600);
+        capabilities.setCapability("browserTimeout", 600);
+        updateQueue = new Timer();
+        updateQueueOfBrowsers();
+    }
+
+    protected void updateQueueOfBrowsers() {
+        timerTask =
+                new TimerTask() {
+                    @Override
+                    public void run() {
+                        try {
+                            List list = getAllNodesList();
+                            LOG.info("Blocking Queue size: " + driversQueue.size());
+                            while (getSessionsCount(list) > 0) {
+                                int size = list.size();
+                                // Check if the queue size is more than the actual
+                                // no of browsers allowed per worker
+                                // means crawler services are idle because all drivers are in queue
+                                if (driversQueue.size() >= (size / noOfWorkers)) {
+                                    SeleniumGridProtocol.Holder holder = driversQueue.take();
+                                    long totalTime = System.currentTimeMillis() - holder.getTime();
+                                    // clear the queue if the total time spent in the queue is more
+                                    // than 4.5 minutes
+                                    // As after 5 mintues the driver would be idle and will throw
+                                    // exception
+                                    // while we try to use that driver for fetching
+                                    if (totalTime > 1000 * 4.5 * 60) {
```

Comment:
5min is the default of Selenium Grid. Can we make it configurable?

Reply:
the 4.5 minutes you want it to be configurable

Reply:
You can configure the default timeout on the server side of selenium grid, so it would be useful to have a config option in SC to adjust it accordingly.

Reply:
> You can configure the default timeout on the server side of selenium grid, so it would be useful to have a config option in SC to adjust it accordingly if it is changed on the grid side.

grid.timeout.minutes this is the configuration I am planning to give